src/propagators/zipkin_b3_text_map_codec.js

```diff
@@ -0,0 +1,157 @@
+                        break;
+                    case ZIPKIN_TRACE_HEADER:
+                        spanContext.traceId = this._decodeValue(carrier[ZIPKIN_TRACE_HEADER]);
+                        break;
+                    case ZIPKIN_SAMPLED_HEADER:
+                        spanContext.flags = spanContext.flags | constants.SAMPLED_MASK;
+                        break;
+                    case ZIPKIN_DEBUG_HEADER:
+                        // "debug implies sampled"
+                        // https://github.com/openzipkin/b3-propagation
+                        spanContext.flags = spanContext.flags | constants.SAMPLED_MASK | constants.DEBUG_MASK;
+                        break;
+                    case constants.JAEGER_DEBUG_HEADER:
+                        spanContext.debugId = this._decodeValue(carrier[constants.JAEGER_DEBUG_HEADER]);
+                        break;
+                    case constants.JAEGER_BAGGAGE_HEADER:
+                        this._parseCommaSeparatedBaggage(baggage, this._decodeValue(carrier[key]));
+                        break;
+                    default:
+                        if (Utils.startsWith(lowerKey, this._baggagePrefix)) {
+                            let keyWithoutPrefix = key.substring(this._baggagePrefix.length);
+                            baggage[keyWithoutPrefix] = this._decodeValue(carrier[key]);
+                        }
+                }
+            }
+        }
+
+        spanContext.baggage = baggage;
+        return spanContext;
+    }
+
+    inject(spanContext: SpanContext, carrier: any): void {
+        carrier[ZIPKIN_TRACE_HEADER] = spanContext.traceIdStr;
+        carrier[ZIPKIN_PARENTSPAN_HEADER] = spanContext.parentIdStr;
+        carrier[ZIPKIN_SPAN_HEADER] = spanContext.spanIdStr;
+
+        // > Since Debug implies Sampled, so don't also send "X-B3-Sampled: 1"
+        // https://github.com/openzipkin/b3-propagation
+
+        if (spanContext.isDebug()) {
+           carrier[ZIPKIN_DEBUG_HEADER] = '1';
+        } else {
+            if (spanContext.isSampled()) {
+                carrier[ZIPKIN_SAMPLED_HEADER] = '1';
+            } else {
+                carrier[ZIPKIN_SAMPLED_HEADER] = '0';
+            }
+        }
+
+        let baggage = spanContext.baggage;
+        for (let key in baggage) {
+            if (baggage.hasOwnProperty(key)) {
+                let value = this._encodeValue(spanContext.baggage[key]);
+                carrier[`${this._baggagePrefix}${key}`] = value;
+            }
+        }
+    }
+
+    // Parse comma separated key=value pair list and add to baggage.
+    // E.g. "key1=value1, key2=value2, key3 = value3"
+    // is converted to map[string]string { "key1" : "value1",
+    //                                     "key2" : "value2",
+    //                                     "key3" : "value3" }
+    _parseCommaSeparatedBaggage(baggage: any, values: string): void {
```

Comment:
maybe refactor it and share with the main jaeger propagator?

Reply:
Sure, sounds good. I wasn't sure how far to dive down the refactoring path just yet.